pkg/sdn/plugin/proxy.go

```diff
@@ -14,12 +14,23 @@ import (
 	"k8s.io/kubernetes/pkg/client/cache"
 	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	pconfig "k8s.io/kubernetes/pkg/proxy/config"
+	ktypes "k8s.io/kubernetes/pkg/types"
 	utilwait "k8s.io/kubernetes/pkg/util/wait"
 )
 
+type firewallItem struct {
+	ruleType osapi.EgressNetworkPolicyRuleType
+	net      *net.IPNet
+}
+
+type policyFirewallItem struct {
+	uid      ktypes.UID
+	firewall []firewallItem
+}
+
 type proxyFirewallItem struct {
-	policy osapi.EgressNetworkPolicyRuleType
-	net    *net.IPNet
+	namespaceFirewalls []policyFirewallItem
```

Comment:
it seems like the code below would be a lot simpler if this was a `map[ktypes.UID]policyFirewallItem` rather than a `[]policyFirewallItem`. (You might even be able to get rid of `policyFirewallItem` and just do `map[ktypes.UID][]firewallItem`?)

Also, maybe type/field naming would help make the code clearer. It's difficult to keep track of all the different "firewalls" and "items"...